### PR TITLE
increase sidecar injector webhook test coverage

### PR DESCRIPTION
**Pull request by ayj**
- add test for annotation replace patching

- add test for upgrading injected sidecar

- benchmark http handler for webhook injection with default
 parameterized template configuration. 

   $ go test -bench .
   BenchmarkInjectServe-4           1000000              1469 ns/op

### CODE REVIEW

File: pilot/pkg/kube/inject/webhook.go [hunk 1]
```diff
@@ -139,7 +139,10 @@ func NewWebhook(configFile, meshFile, certFile, keyFile string, port int) (*Webh
 		meshFile:               meshFile,
 		watcher:                watcher,
 	}
-	http.HandleFunc("/inject", wh.serveInject)
+	h := http.NewServeMux()
```
Comment (cmluciano) [inline pilot/pkg/kube/inject/webhook.go]: Do we do this to capture more metrics? I'm not sure what the desired change is here.

File: pilot/pkg/kube/inject/webhook_test.go [hunk 2]
```diff
@@ -230,6 +243,290 @@ func TestWebhookInject(t *testing.T) {
...
+
+	var (
+		configFile = filepath.Join(dir, "config-file.yaml")
+		meshFile   = filepath.Join(dir, "mesh-file.yaml")
+		certFile   = filepath.Join(dir, "cert-file.yaml")
+		keyFile    = filepath.Join(dir, "key-file.yaml")
+		port       = 0
+	)
+
+	if err := ioutil.WriteFile(configFile, configBytes, 0644); err != nil { // nolint: vetshadow
```
Comment (cmluciano) [inline pilot/pkg/kube/inject/webhook_test.go]: Do we clean these files up or rely on normal tmp dir cleanup from the OS?

Comment (ayj) [reply pilot/pkg/kube/inject/webhook_test.go]: We do now. Just added code to remove the temp directory when the test completes or errors out.

File: pilot/pkg/kube/inject/webhook.go (hunk 1 above)
Comment (ayj) [reply pilot/pkg/kube/inject/webhook.go]: The original code panic'd when I added benchmarks because the default ServeMux is shared across tests.

**Comment by andraxylia** [comment]
e2e-smoke fails because of #3320

**Comment by andraxylia** [comment]
pilot-e2e-fails for http reachability in auth enabled mode, could be due to the slow loading of certificates #3139 , but we are missing the logs we cannot be sure. In any case, the failure is unrelated to the changes in this PR.

**Pull request merged**